### DISCUSSION TITLE
How can I operate the extension in only ANSI C mode?

### DISCUSSION THREAD

**Discussion by ghost** [microsoft/vscode-cpptools#9533] [Q&A]
Greetings,

I am getting red, squiggly error reports that pertain to C++ in a C11 code base. For instance,

```
inline uint8_t bytes_pack_sse2(const uint8_t a[static 8])
```
gives an error

```
static array size is a C99 feature, not permitted in C++ [Semantic Issue]Clang
```

Is there a way to tell the plugin to only use ANSI suggestions in some files?

**Comment by sean-mcmanus** [comment]
Where is that error coming from? It doesn't appear to be from our extension. Something seems to be interpreting the file as C++ instead of C. What is the file's extension? What language is shown in the VS Code status bar in the bottom right? Are there any files.associations set?

**Comment by ghost** [comment]
>Where is that error coming from?

Is there a way to tell from the `Problems` tab which extension is throwing the problem?

>What is the file's extension?

.c

>What language is shown in the VS Code status bar in the bottom right?

![image](https://user-images.githubusercontent.com/77119510/176978048-6c29676d-7ba7-4395-a66a-17e4b3b09c78.png)

EDIT: I am realizing that it's showing a configuration and not a language. Here is the config from c_cpp_properties

```
{
	"configurations": [
		{
			"name": "clang",
			"includePath": [
				"${workspaceFolder}/include",
				"${workspaceFolder}/lib/genii"
			],
			"intelliSenseMode": "linux-clang-x64",
			"compilerPath": "/home/james/base/bin/clang",
			"cStandard": "c11",
			"cppStandard": "c++11"
		}
	],
	"version": 4
}
```

>Are there any files.associations set?

The only file associations I have are
```	
"files.associations": {
	"*.h": "c"
},
```

**Reply by sean-mcmanus** [reply]
Our Problems have "C/C++(id)" as the source and error code id: 
![image](https://user-images.githubusercontent.com/19859882/176978283-f78ef74a-13dd-4eab-94f6-3ec73e50bbf4.png)

**Reply by ghost** [reply]
Ah, yes it looks like it is not your extension

![image](https://user-images.githubusercontent.com/77119510/176978420-d8d48d90-e64b-419c-b1d7-ed65477cd215.png)

**Reply by ghost** [reply]
Found that it was coming from the C/C++ advanced lint extension. Sorry for the noise. Thanks for the help.

**Reply by ghost** [reply]
https://github.com/jbenden/vscode-c-cpp-flylint

Was coming from this
